mindmeld/components/question_answerer.py

```diff
@@ -133,1121 +162,2704 @@ def get(self, index, size=10, query_type=None, **kwargs):
+
+        index_name = self._resolve_deprecated_index_name(index_name, kwargs.pop("index", None))
+        return self._get(index=index_name, size=size, query_type=query_type,
+                         app_namespace=app_namespace, **kwargs)
+
+    def build_search(self, index_name=None, ranking_config=None, app_namespace=None, **kwargs):
+        """Build a search object for advanced filtered search.
+
+        Args:
+            index_name (str): index name of knowledge base object.
+            ranking_config (dict, optional): overriding ranking configuration parameters.
+            app_namespace (str, optional): The namespace of the app. Used to prevent
+                collisions between the indices of this app and those of other apps.
+        Returns:
+            Search: a Search object for filtered search.
+        """
+
+        index_name = self._resolve_deprecated_index_name(index_name, kwargs.pop("index", None))
+        return self._build_search(index=index_name, ranking_config=ranking_config,
+                                  app_namespace=app_namespace, **kwargs)
+
+    def load_kb(self, index_name, data_file, **kwargs):
+        """Loads documents from disk into the specified index in the knowledge
+        base. If an index with the specified name doesn't exist, a new index
+        with that name will be created in the knowledge base.
+
+        Args:
+            index_name (str): The name of the new index to be created.
+            data_file (str): The path to the data file containing the documents
+                to be imported into the knowledge base index. It could be
+                either json or jsonl file.
+
+        Optional Args (used by all QA classes):
+            app_namespace (str): The namespace of the app. Used to prevent
+                collisions between the indices of this app and those of other apps.
+            clean (bool): Set to true if you want to delete an existing index
+                and reindex it
+            embedding_fields (list): List of embedding fields can be directly passed in
+                instead of adding them to QA config
+
+        Optional Args (Elasticsearch specific):
+            es_host (str): The Elasticsearch host server.
+            es_client (Elasticsearch): The Elasticsearch client.
+            connect_timeout (int): The amount of time for a connection to the Elasticsearch host.
+        """
+
+        if (
+            ("config" in kwargs and kwargs["config"]) or
+            ("app_path" in kwargs and kwargs["app_path"])
+        ):
+            msg = f"Passing 'config' or 'app_path' to '.load_kb()' method is no longer " \
+                  f"supported. Create a {self.__class__.__name__} instance with the required " \
+                  f"configurations and/or app path before calling '.load_kb()'."
+            raise ValueError(msg)
+        self._load_kb(index_name, data_file, **kwargs)
+
+
+class ResolverBasedQuestionAnswerer(BaseQuestionAnswerer):
```

Comment:
I think ResolverBased is confusing to the end-user (if they ever have to debug something) compared to NonElasticSearch. Should we rename it to NativeQuestionAnswerer or something that indicates the ranking happens locally in some sense? @mhuang2173 @vijay120 @murali1996

Reply:
Good point! I'm in for naming it as `NativeQuestionAnswerer`; if we don't have any other suggestions I can go ahead and rename it that way. @mhuang2173 @vijay120 @vembar

Reply:
I think `NativeQuestionAnswerer` is great since it reflects that it uses native ranking implementation!